src/Http/Headers/src/BaseHeaderParser.cs

```diff
@@ -14,9 +15,9 @@ protected BaseHeaderParser(bool supportsMultipleValues)
 
         protected abstract int GetParsedValueLength(StringSegment value, int startIndex, out T parsedValue);
 
-        public sealed override bool TryParseValue(StringSegment value, ref int index, out T parsedValue)
+        public sealed override bool TryParseValue(StringSegment value, ref int index, [MaybeNullWhen(false)]out T parsedValue)
```

Comment:
It looks like TryParseValue can return true even in cases where parsedValue is never set if SupportsMultipleValues is also true.